app/grandchallenge/algorithms/urls.py

```diff
@@ -35,6 +36,7 @@
         ComponentInterfaceList.as_view(),
         name="component-interface-list",
     ),
+    path("not-found/", AlgorithmNotFound.as_view(), name="no-slug"),
```

Comment:
Lost the comment but saw the 404 template was updated, is this route still needed? If not, remove the view too,